cherab/solps/formats/balance.py

```diff
@@ -0,0 +1,192 @@
+    inside_outside = AxisymmetricMapper(Discrete2DMesh(mesh.vertex_coords, mesh.triangles, inside_outside_data, limit=False))
+    sim._inside_mesh = inside_outside
+	
+    # Load the neutrals data
+    if 'D0' in sim.species_list:
+        for i in arange(len(sim.species_list)):
+            if sim.species_list[i] == 'D0':
+                D0_indx = i
+
+    # Replace the deuterium neutrals density (from the fluid neutrals model by default) with
+    # the values calculated by EIRENE - do the same for other neutrals?
+    if 'dab2' in fhandle.variables.keys():
+        sim.species_density[:,:,D0_indx] = copy.deepcopy(fhandle.variables['dab2'].data[0,:,0:-2])
+        eirene_run = True
+    else:
+        eirene_run = False
+    
+    # Calculate the total radiated power
+    
+    
+    if eirene_run is True:
+        # Total radiated power from B2, not including neutrals
+        b2_ploss = fhandle.variables['b2stel_she_bal'].data/vol
+        
+        # Electron energy loss due to interactions with neutrals
+        if 'eirene_mc_eael_she_bal' in fhandle.variables.keys():
+            eirene_ecoolrate = sum(fhandle.variables['eirene_mc_eael_she_bal'].data,axis=0)/vol
+            
+        # Ionisation rate from EIRENE, needed to calculate the energy loss to overcome the ionisation potential of atoms
+        if 'eirene_mc_papl_sna_bal' in fhandle.variables.keys():
+            eirene_potential_loss = 13.6*sum(fhandle.variables['eirene_mc_papl_sna_bal'].data,axis=(0))[1,:,:]*Q/vol
+        
+        # This will be negative (energy sink); take absolute valu
+        sim._total_rad = abs(b2_ploss+(eirene_ecoolrate-eirene_potential_loss))
+        
+    else:
+        #print(b2_ploss.shape)
+        #print(cr_x.shape)
+        #print(sum(b2_ploss,axis=0).shape)
+        
+        # Total radiated power from B2, not including neutrals
+        b2_ploss = sum(fhandle.variables['b2stel_she_bal'].data,axis=0)/vol
+        
+        potential_loss = sum(fhandle.variables['b2stel_sna_ion_bal'].data,axis=0)/vol
+        
+        #rad_dens = (-sum(b2stel_she_bal,3)+13.6*Q*b2stel_sna_ion_bal[:,:,0])/vol
+        
+        sim._total_rad = abs(13.6*Q*potential_loss-b2_ploss)
+	
+    fhandle.close()	
+
+    return sim
+
+
+def load_mesh_from_files(mesh_file_path, debug=False):
+    """
+    Load SOLPS grid description from B2 Eirene output file.
+
+    :param str filepath: full path for B2 eirene mesh description file
+    :param bool debug: flag for displaying textual debugging information.
+    :return: tuple of dictionaries. First is the header information such as the version, label, grid size, etc.
+      Second dictionary has a ndarray for each piece of data found in the file.
+    """
+    _, _, geom_data_dict = load_b2f_file(mesh_file_path, debug=debug)
+
+    cr_x = geom_data_dict['crx']
+    cr_z = geom_data_dict['cry']
+    vol = geom_data_dict['vol']
+
+    # build mesh object
+    return SOLPSMesh(cr_x, cr_z, vol)
```

Comment:
This is identical to a function with the same name in `raw_simulation_files.py`. Overzealous copy and paste?

Reply:
Exactly right, this function has been removed